src/test/socket/connect/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+add_test(NAME connect COMMAND ../../target/debug/test_connect)
+foreach(Method ptrace preload)
+    # runs the shadow tests with only the passing tests, not all tests
+    add_test(NAME connect-shadow-${Method} COMMAND ${CMAKE_BINARY_DIR}/src/main/shadow --interpose-method=${Method}
+	    -l debug -d connect.shadow-${Method}.data ${CMAKE_CURRENT_SOURCE_DIR}/connect.shadow.config.xml)
+endforeach()
+
+set(TestNames connect connect-shadow-ptrace connect-shadow-preload)
+set_property(TEST ${TestNames} PROPERTY ENVIRONMENT "RUST_BACKTRACE=1")
+set_property(TEST ${TestNames} PROPERTY FAIL_REGULAR_EXPRESSION "main error code '.*' for process")
```

Comment:
Is this to catch the case where Shadow killed the process but exited with a 0 status? Probably worth a brief explanation and reference to the issue.

Reply:
Added.